Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,900 @@
+        }
+
+        if (other.contextMenuLuaActions != null)
+        {
+            contextMenuLuaActions = new List<ContextMenuLuaAction>(other.contextMenuLuaActions);
+        }
+
+        getSpriteNameAction = other.getSpriteNameAction;
+
+        if (other.PowerConnection != null)
+        {
+            PowerConnection = other.PowerConnection.Clone() as Connection;
+            World.Current.PowerNetwork.PlugIn(PowerConnection);
+            PowerConnection.NewThresholdReached += OnNewThresholdReached;
+        }
+
+        if (other.funcPositionValidation != null)
+        {
+            funcPositionValidation = (Func<Tile, bool>)other.funcPositionValidation.Clone();
+        }
+
+        tileTypeBuildPermissions = other.tileTypeBuildPermissions;
+
+        LocalizationCode = other.LocalizationCode;
+        UnlocalizedDescription = other.UnlocalizedDescription;
+    }
+
+    /// <summary>
+    /// This event will trigger when the furniture has been changed.
+    /// This is means that any change (parameters, job state etc) to the furniture will trigger this.
+    /// </summary>
+    public event Action<Utility> Changed;
+
+    /// <summary>
+    /// This event will trigger when the furniture has been removed.
+    /// </summary>
+    public event Action<Utility> Removed;
+
+    /// <summary>
+    /// This event will trigger if <see cref="IsOperating"/> has been changed.
+    /// </summary>
+    public event Action<Utility> IsOperatingChanged;
+
+    /// <summary>
+    /// Gets the width of the furniture.
+    /// </summary>
+    public int Width 
+    { 
+        get { return 1; }
+    }
+
+    /// <summary>
+    /// Gets the height of the furniture.
+    /// </summary>
+    public int Height 
+    { 
+        get { return 1; }
+    }
+
+    /// <summary>
+    /// Gets the tint used to change the color of the furniture.
+    /// </summary>
+    /// <value>The Color of the furniture.</value>
+    public Color Tint { get; private set; }
+
+    /// <summary>
+    /// Gets the spot where the Character will stand when he is using the furniture. This is relative to the bottom
+    /// left tile of the sprite. This can be outside of the actual furniture.
+    /// </summary>
+    /// <value>The spot where the Character will stand when he uses the furniture.</value>
+    public Vector2 JobSpotOffset { get; private set; }
+
+    /// <summary>
+    /// Gets the EventAction for the current furniture.
+    /// These actions are called when an event is called. They get passed the furniture
+    /// they belong to, plus a deltaTime (which defaults to 0).
+    /// </summary>
+    /// <value>The event actions that is called on update.</value>
+    public EventActions EventActions { get; private set; }
+
+    /// <summary>
+    /// Gets the Connection that the furniture has to the power system.
+    /// </summary>
+    /// <value>The Connection of the furniture.</value>
+    public Connection PowerConnection { get; private set; }
```

Comment:
When you copy-paste result should have some logical appearance.
No nead for `Connection` for this